### PR TITLE
feat: implement isolation

### PR DESCRIPTION
**Pull request by nchietala**

### CODE REVIEW

**Comment by nchietala** [comment]
I'm done with this and I'd like to merge it while it's still only a medium PR not a large one. Now I will work on actually getting a read on the odo type meters before I go about doing the dials.

**Comment by nchietala** [comment]
actually, nevermind, I never got the dial isolate filter working. I'll do that before I get this merged.

File: OpenMTRDemo/Filters/IsolateDialsFilter.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                    Mat output = new Mat();
+                    Cv2.GetRectSubPix(image, new Size(diameter, diameter), center, output);
+                    Editor.Dials[i] = new MeterImage("", output);
+                }
+            }
+        }
+
+        private void SortCircles()
+        {
+            List<CircleSegment> unsortedList = new List<CircleSegment>(_circleList.ToArray());
```
Comment (cclinard84) [inline OpenMTRDemo/Filters/IsolateDialsFilter.cs]: Any way we could sort this list using linq? Something similar to https://github.com/OpenMTR/OpenMTR/blob/master/OpenMTR/Code/Odometer.cs#L75

File: OpenMTRDemo/Filters/IsolateDialsFilter.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            InitializeComponent();
+        }
+
+        public override void ApplyFilter(Mat image)
+        {
+            if (image.Type() == MatType.CV_8UC1)
+            {
+                _circleList = new List<CircleSegment>();
+                CircleSegment[] circles = Cv2.HoughCircles(image, HoughMethods.Gradient, 1, (Math.Min(image.Height, image.Width) / 12), 250, 100, (Math.Min(image.Height, image.Width) / 20), (Math.Min(image.Height, image.Width) / 3));
+                foreach (CircleSegment circle in circles)
```
Comment (cclinard84) [inline OpenMTRDemo/Filters/IsolateDialsFilter.cs]: I feel like this logic can be simplified quite a bit by going ahead and checking to make sure the center XY coordinates are not the same throughout the list and then proceed to create the boundaries.

File: OpenMTRDemo/Filters/IsolateDialsFilter.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+        public override void Cascade(Mat image)
+        {
+            if (_circleList.Count == 4)
+            {
+                Editor.Dials = new MeterImage[4];
+                SortCircles();
+                for (int i = 0; i < 4; i++)
+                {
+
```
Comment (cclinard84) [inline OpenMTRDemo/Filters/IsolateDialsFilter.cs]: A wild NewLine has appeared!

**Pull request merged**